### PR TITLE
feat(ticket-page): implement dynamic cards for ticket page #88

### PR DESCRIPTION
**Pull request by Vaibhavsahu2810**
## Overview

1. This PR fixes or fixes part of #88 .
2. This PR does the following: This pull request introduces significant changes and new features related to the ticketing system in both the backend and frontend of the application. The key enhancements include modifying backend services and controllers, seeding data according to a new schema, fetching and displaying ticket data dynamically on the frontend, and adding filtering functionality.

## Essential Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] "Allow edits from maintainers" is checked. (See [here](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) for instructions on how to enable it.)
- [x] The PR is made from a branch that's **not** called "main/master".

## Proof that changes are correct
https://www.loom.com/share/14777e7eb2c94aaa8022051b1a7bde4c?sid=10910a6d-4717-4e03-b0c7-86729946abd5
Above i have provided the video link as the proof of my changes
### Changes in Backend

1. **Modified `ticket.service.ts`**:
   - Updated the service methods to align with the new schema.
   - Implemented logic to create and manage tickets with the new attributes, ensuring proper interaction with the database.

2. **Modified `ticket.controller.ts`**:
   - Adjusted the endpoints to accommodate the changes in the service layer.
   - Added new endpoints or updated existing ones to support fetching, creating, and managing ticket data as per the new schema.

3. **Updated `seed.mjs`**:
   - Updated the seed data script to reflect the new schema.
   - Ensured the seeded data is consistent and includes all necessary attributes for testing and development purposes.

### Changes in Frontend

1. **Fetched Ticket Data**:
   - Implemented API calls to fetch ticket data from the backend URL.
   - Ensured data is retrieved efficiently and handled any potential errors gracefully.

2. **Ticket Component**:
   - Created or updated the ticket component to display ticket data dynamically.
   - Utilized the fetched data to populate the component, ensuring it reflects the latest ticket information.

3. **Profile Page Integration**:
   - Integrated the ticket component into the profile page.
   - Ensured that the user's tickets are displayed correctly and updated dynamically as needed.

4. **Filtering Functionality**:
   - Added the ability to filter ticket data based on user needs.
   - Implemented various filtering options, such as by date, status, or other relevant criteria.
   - Ensured the filtering is efficient and provides a seamless user experience.

## PR Pointers

- If you need a review or an answer to a question, and don't have permissions to assign people, **leave a comment** like the following: "{{Question/comment}} @{{reviewer_username}} PTAL".
- Never force push. If you do, your PR will be closed.

### CODE REVIEW

**Comment by RISHIKESHk07** [comment]
@Vaibhavsahu2810 please make sure prisma schema you are following is same as @rohansen856 once as well , code looks fine anyway , @Pratik2026 any suggestions ?

**Comment by Vaibhavsahu2810** [comment]
@RISHIKESHk07 i have successfully verified the schema with @rohansen856

**Comment by RISHIKESHk07** [comment]
@Vaibhavsahu2810 i see a few small conflicts , please fix it and maybe show a demo of the working of the page as well

**Comment by Vaibhavsahu2810** [comment]
> @Vaibhavsahu2810 i see a few small conflicts , please fix it and maybe show a demo of the working of the page as well

@RISHIKESHk07 successfully resolved the conflicts and here is the proof of work
https://www.loom.com/share/8ea4380ca836449285158a497f94e849?sid=3906a217-3a63-4cf7-acec-1d7acc0ed7ca

File: backend/prisma/seed.mjs [hunk 1]
```diff
@@ -72,7 +73,54 @@ const demoData = {
         'Saturday',
         'Sunday',
       ],
-      ticketPrice: 20,
+    },
+  ],
+  users: [
```
Comment (Pratik2026) [inline backend/prisma/seed.mjs:78]: @Vaibhavsahu2810 we already have users, tickets, and orders keys, yet you have added the objects by creating their duplicate keys. Can you explain the reason?

**Review by Pratik2026 — COMMENTED** [review]
@Vaibhavsahu2810, Are you sure that seeding is working in your system because when I am trying to seed the DB, all I am getting is a bunch of errors.

**Comment by Pratik2026** [comment]
Some errors are due to duplicate keys and incorrect naming(when accessing demo data). Also, about the `foreign key constraint` error, please make sure that the id defined in the orders demodata refers to the existing id of schedule demodata.

**Comment by Pratik2026** [comment]
Once you have resolved the seeeding, look into the invalid time value error as the ticket is not able to be created in the backend.
Also please resolve these two issues in a separate commit, one for seeding and another for invalid time value error

**Comment by Vaibhavsahu2810** [comment]
@Pratik2026 i have fixed the seed data and invalid time error , also i have updated the frontend to dynamically use the schedule id as earlier is was hardcoded
here is the proof of work https://www.loom.com/share/3e2456854113415c98bbe3dcb4496eaf?sid=f2df53e0-636a-42d8-969f-8dac37b41eb3

**Comment by Pratik2026** [comment]
@Vaibhavsahu2810 seeding is still not working, You have not resolved the foreign key constraint error. Try running npm run seed separately in backend folder, I guess you have been running all three commands simultaneously that's why you have not notice the error yet

**Comment by Vaibhavsahu2810** [comment]
@Pratik2026 fixed the issue of foreign key constraint by changing the order of deletion

**Comment by Pratik2026** [comment]
LGTM.

**Comment by Pratik2026** [comment]
merged, thanks for the contribution @Vaibhavsahu2810

**Pull request merged**